rust/module.rs

```diff
@@ -244,18 +270,59 @@ pub fn module(ts: TokenStream) -> TokenStream {
             &param_name,
             &param_description,
         ));
+        let param_type_internal = match param_type.as_ref() {
+            "str" => "*mut u8",
+            "invbool" => "bool",
+            _ => &param_type,
+        };
+        let param_default = match param_type.as_ref() {
+            "str" => format!("b\"{}\0\" as *const u8 as *mut u8", param_default),
+            _ => param_default,
+        };
+        let read_func = match param_type.as_ref() {
+            "str" => format!(
+                "
+                    fn read(&self) -> Result<&str, core::str::Utf8Error> {{
+                        unsafe {{
```

Comment:
If you can put a `// SAFETY: reason` comment, it would be great (I know we don't do it now in many places, but it seems a good policy from the stdlib to require every `unsafe` block to have an explanation).